packages/web-pkg/src/helpers/share/link.ts

```diff
@@ -180,3 +160,24 @@ export const createQuicklink = async (args: CreateQuicklink): Promise<Share> =>
     storageId: resource.fileId || resource.id
   })
 }
+
+export const getDefaultLinkPermissions = ({
+  ability,
+  store
+}: {
+  ability: Ability
+  store: Store<any>
+}) => {
+  const canCreatePublicLink = ability.can('create-all', 'PublicLink')
+  if (!canCreatePublicLink) {
+    return SharePermissionBit.Internal
+  }
+
+  let defaultPermissions: number =
+    store.state.user.capabilities.files_sharing?.public?.default_permissions
+  if (defaultPermissions === undefined) {
+    defaultPermissions = SharePermissionBit.Read
+  }
```

Comment:
Defining the same default as fallback here as in `useCapabilityFilesSharingPublicDefaultPermissions` is a shortcoming of our architecture, see https://github.com/owncloud/web/issues/9748.